### ISSUE TITLE
Maven is not packaging Robotium.jar into apk for testing???

### ISSUE THREAD

**Issue by renas** [RobotiumTech/robotium#92]
_From [john.ar...@gmail.com](https://code.google.com/u/107363567412177313294/) on August 01, 2010 13:58:48_

What steps will reproduce the problem? 1. Setup mavenized main project as per maven-android-eclipse docs. 
2. Setup mavenized test project as per maven-android-eclipse docs.
3. Setup Robotium dependency on test project. What is the expected output? What do you see instead? I expect a Junit runner result - not an exception of a missing class.
Seems like the apk doesn't have the source packed in for some reason.
This is despite having 0 compile time issues. What version of the product are you using? On what operating system? mvn -version
Apache Maven 2.2.1 ( r801777 ; 2009-08-06 15:16:01-0400)
Java version: 1.6.0_20
Java home: /usr/java/jdk1.6.0_20/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux" version: "2.6.18-194.8.1.el5" arch: "amd64" Family: "unix" Please provide any additional information below. DDMS Output

08-01 17:48:36.505: ERROR/dalvikvm(406): Could not find class 'com.jayway.android.robotium.solo.Solo', referenced from method com.overlord.test.NotificationTest.setUp
08-01 17:48:36.505: WARN/dalvikvm(406): VFY: unable to resolve new-instance 16 (Lcom/jayway/android/robotium/solo/Solo;) in Lcom/overlord/test/NotificationTest;
08-01 17:48:36.505: WARN/dalvikvm(406): VFY:  rejecting opcode 0x22 at 0x0000
08-01 17:48:36.505: WARN/dalvikvm(406): VFY:  rejected Lcom/overlord/test/NotificationTest;.setUp ()V
08-01 17:48:36.505: WARN/dalvikvm(406): Verifier rejected class Lcom/overlord/test/NotificationTest;
08-01 17:48:36.505: DEBUG/AndroidRuntime(406): Shutting down VM
08-01 17:48:36.505: WARN/dalvikvm(406): threadid=3: thread exiting with uncaught exception (group=0x4001aa28)
08-01 17:48:36.505: ERROR/AndroidRuntime(406): Uncaught handler: thread main exiting due to uncaught exception
08-01 17:48:36.515: ERROR/AndroidRuntime(406): java.lang.VerifyError: com.overlord.test.NotificationTest
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at java.lang.Class.getDeclaredConstructors(Native Method)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at java.lang.Class.getConstructor(Class.java:484)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at junit.framework.TestSuite.getTestConstructor(TestSuite.java:177)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at junit.framework.TestSuite.<init>(TestSuite.java:59)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at junit.runner.BaseTestRunner.getTest(BaseTestRunner.java:103)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.test.AndroidTestRunner.getTest(AndroidTestRunner.java:124)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.test.AndroidTestRunner.setTestClassName(AndroidTestRunner.java:52)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.test.suitebuilder.TestSuiteBuilder.addTestClassByName(TestSuiteBuilder.java:81)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.test.InstrumentationTestRunner.parseTestClass(InstrumentationTestRunner.java:375)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.test.InstrumentationTestRunner.parseTestClasses(InstrumentationTestRunner.java:357)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.test.InstrumentationTestRunner.onCreate(InstrumentationTestRunner.java:325)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:3848)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.app.ActivityThread.access$2800(ActivityThread.java:116)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1831)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.os.Handler.dispatchMessage(Handler.java:99)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.os.Looper.loop(Looper.java:123)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at android.app.ActivityThread.main(ActivityThread.java:4203)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at java.lang.reflect.Method.invokeNative(Native Method)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at java.lang.reflect.Method.invoke(Method.java:521)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:791)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:549)
08-01 17:48:36.515: ERROR/AndroidRuntime(406):     at dalvik.system.NativeStart.main(Native Method)
08-01 17:48:36.525: INFO/Process(52): Sending signal. PID: 406 SIG: 3
08-01 17:48:36.525: WARN/ActivityManager(52): Error in app com.overlord running instrumentation ComponentInfo{com.overlord.test/android.test.InstrumentationTestRunner}:
08-01 17:48:36.525: WARN/ActivityManager(52):   com.overlord.test.NotificationTest
08-01 17:48:36.525: WARN/ActivityManager(52):   java.lang.VerifyError: com.overlord.test.NotificationTest

Eclipse - Run Android JUnit Test output

Process crash is using the APK style test initialization,
Test run fail is using the standard test initialization.

[2010-08-01 13:48:28 - Overlordtest] Launching instrumentation android.test.InstrumentationTestRunner on device emulator-5554
[2010-08-01 13:48:30 - Overlordtest] Collecting test information
[2010-08-01 13:48:33 - Overlordtest] Test run failed: Process crashed.
[2010-08-01 13:48:37 - Overlordtest] ------------------------------
[2010-08-01 13:48:37 - Overlordtest] Android Launch!
[2010-08-01 13:48:37 - Overlordtest] adb is running normally.
[2010-08-01 13:48:37 - Overlordtest] Performing android.test.InstrumentationTestRunner JUnit launch
[2010-08-01 13:48:37 - Overlordtest] Automatic Target Mode: using existing emulator 'emulator-5554' running compatible AVD 'googleapi4'
[2010-08-01 13:48:39 - Overlordtest] Application already deployed. No need to reinstall.
[2010-08-01 13:48:39 - Overlordtest] Project dependency found, installing: Overlord
[2010-08-01 13:48:40 - Overlord] Application already deployed. No need to reinstall.
[2010-08-01 13:48:40 - Overlordtest] Launching instrumentation android.test.InstrumentationTestRunner on device emulator-5554
[2010-08-01 13:48:42 - Overlordtest] Collecting test information
[2010-08-01 13:48:44 - Overlordtest] Test run failed: com.overlord.test.NotificationTest

Primary POM.

	\<artifactId>overlord</artifactId>
	\<packaging>apk</packaging>
	\<name>Overlord</name>
	\<version>0.0.1-SNAPSHOT</version>

	\<dependencies>
		\<dependency>
			\<groupId>android</groupId>
			\<artifactId>android</artifactId>
			\<version>2.1_r2</version>
			\<scope>provided</scope>
		\</dependency>
		\<dependency>
			\<groupId>com.google.android.maps</groupId>
			\<artifactId>maps</artifactId>
			\<version>7_r1</version>
			\<scope>provided</scope>
		\</dependency>
		\<dependency>
			\<groupId>junit</groupId>
			\<artifactId>junit</artifactId>
			\<version>4.8.1</version>
		\</dependency>
	\</dependencies>
	
	\<build>
		\<sourceDirectory>src</...

_Original issue: http://code.google.com/p/robotium/issues/detail?id=13_

**Comment by renas** [comment]
_From [hugo.josefson.old@gmail.com](https://code.google.com/u/hugo.josefson.old@gmail.com/) on August 12, 2010 06:51:54_

Hi John!

I have spent some time now trying to reproduce your error. Using your partial pom snippets above, I have recreated a two-module project which uses your exact parameters. Everything works fine with Maven.

Here is the code: http://github.com/hugojosefson/robotium-issue13 You can run 'mvn install' from the root of that code repository, and it works. It runs the tests which depend on Robotium.

Note that when you set up your project with Maven, you should build it with Maven. IDE's such as IntelliJ IDEA and Eclipe may or may not do the things you expect. Maven is always the stable constant that you can rely on to build according to the pom.xml.

Since you only posted parts of your poms, and no code nor project contents, I guessed what the projects contained. (That is also the reason why this took so long; I had to set aside time to re-create these projects. I apologise for the wait.) If your projects differ from what I recreated, please fork my repo on github and commit+push changes to it so it exhibits the error you reported. Or simply attach a zip file to this issue with everything needed to build it and reproduce the error.

On the other hand, if you have found a solution, please let us know so we can close this issue!

Thanks,
Hugo

**Status:** Incomplete  
  **Owner:** h...@josefson.org

**Comment by renas** [comment]
_From [john.ar...@gmail.com](https://code.google.com/u/107363567412177313294/) on August 12, 2010 17:35:05_

Hi Hugo,

I'm sorry, I thought I'd get some requests for more information, I will attached the Test project and the Apk and try to describe the setup in more details.

No, I haven't solved it yet.

I'll update this soon.

Thanks for your time.

**Comment by renas** [comment]
_From [john.ar...@gmail.com](https://code.google.com/u/107363567412177313294/) on August 12, 2010 22:23:08_

Hi,

I've spent some time analyzing your changes and the differences between my setup and structure. It seems to me that the way I named the test packages were incorrect.

I loaded your project off git, ran perfectly - just replaced the submodules and spent some tiem using trial and error. 

The test package was overlord.test, instead of being inside the main overlord package which required me to import the source package. I think that was the screw up.

Everything works perfectly now.
